### ISSUE TITLE
Restructure `HDF5DataSet` and `Trainer` classes

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#236]
### HDF5DataSet:
- Use a more generic name (Dataset?)
  - Settled on GraphDataset because pytorch-geometric already has a Dataset class
- Initialize with:
  - hdf5 path of raw data
  - keys of the hdf5 file that needs to be included in the specific HDF5DataSet instance
  - filtering options for features and/or targets
  - it should have `torch_geometric.data.data` Data object instance as an attribute

### Trainer
- Initialized with:
  - batch size, epochs, shuffle, ...
  - callback (early stopping)
  - metrics exporters
  - saving options
  - pretrained model
- Include a training method (called fit or train), that takes inputs
  - features
  - initialized neural network (Net)
  - HDF5Datasets 
  - targets/task

rename files to all lower case letters (as per python convention)

**Comment by DaniBodor** [comment]
features, targets, etc need to be in Dataset class, due to the pay pytorch gemoetric handles these. Basically, the get function doesn't work as intended if the Dataset class contains features that are not used for training. Hence, the feature selection needs to be done on the Dataset class already.

Will make separate issue for moving attributes from Trainer class to Trainer.fit (or Trainer.train?) method: #234
